### PR TITLE
Search for items by group

### PR DESCRIPTION
**Pull request by jgondron**
## DLTP-1358: Search for items by group

ff72c667e49d2d5e0094a26407cc2ee52215c573

- Added a facet for the groups that are available within the resultant set.
- Added a helper method to curate_helper to map group ids to the more user friendly group names
- Changed the "View All Content/My Content" from a form with radio buttons to links so that I could reconstruct the "All works" link to exclude any group filters
- Added seed data for two different groups for testing
- Updated README to instruct how to create the docker image with seed data pre-loaded

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -43,6 +43,11 @@ You can also run Fedora and Solr via Docker:
 $ docker run -p 8983:8983 -d --name curate-jetty -t ndlib/curate-jetty
 ```
 
+Or with pre-generated development seed data:
```
Comment (jeremyf) [inline README.md]: Huzzah! And then some!

File: app/controllers/catalog_controller.rb [hunk 2]
```diff
@@ -218,6 +247,20 @@ def departments
...
+  # show it when the user is viewing "My Works". This removes the facet when not viewing "My Works".
+  # If there is a way to dynamically add it to the config based on a user param and specify the order,
+  # then we can invert this logic to add instead of remove
+  def remove_group_facets(solr_parameters, user_params)
+    only_mine = (user_params["works"] == 'mine')
+    if !only_mine
+      # Remove the groups facet
+      user_params["f"].delete "edit_access_group_ssim" if user_params["f"] && user_params["f"].include?("edit_access_group_ssim")
+      blacklight_config.facet_fields.delete "edit_access_group_ssim" if blacklight_config.facet_fields && blacklight_config.facet_fields.include?("edit_access_group_ssim")
+      #solr_parameters[:"facet.field"].select! { |field| field != "{!ex=edit_access_group_ssim}edit_access_group_ssim" }
```
Comment (jeremyf) [inline app/controllers/catalog_controller.rb]: Should this be commented out? If so, please remove.

File: app/helpers/curate_helper.rb [hunk 3]
```diff
@@ -17,6 +17,17 @@ def collection_title_from_pid(value)
     c.nil? ? value : c.title
   end
 
+  # Gets the user friendly group title by id. Note: This implementation does not support multiples,
+  # ex: "und:groupid1 OR und:groupid2"
+  def group_name_from_pid(value)
+    begin
+      p = Hydramata::Group.load_instance_from_solr(value)
```
Comment (jeremyf) [inline app/helpers/curate_helper.rb]: Please rename the `p` variable to `group` (or something that gives it more meaning than `p`.

Comment (jgondron) [reply app/helpers/curate_helper.rb]: But, p is for perfect. The most perfect variable name...

**Pull request merged**